vcd/resource_vcd_org_vdc_template_instance.go

```diff
@@ -0,0 +1,97 @@
+package vcd
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+)
+
+func resourceVcdOrgVdcTemplateInstance() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdVdcTemplateInstantiateCreate,
+		ReadContext:   resourceVcdVdcTemplateInstantiateRead,
+		DeleteContext: resourceVcdVdcTemplateInstantiateDelete,
+		Schema: map[string]*schema.Schema{
+			"org_vdc_template_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "ID of the VDC template to instantiate",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the VDC to be instantiated",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Description of the VDC to be instantiated",
+			},
+			"org_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "ID of the Organization where the VDC will be instantiated",
+			},
+		},
+	}
+}
+
+func resourceVcdVdcTemplateInstantiateCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vdcTemplateId := d.Get("org_vdc_template_id").(string)
+	vdcTemplate, err := meta.(*VCDClient).GetVdcTemplateById(vdcTemplateId)
+	if err != nil {
+		return diag.Errorf("could not instantiate the VDC Template: %s", err)
```

Comment:
Should the error not be something like could not look up VDC template?

Reply:
Yes, that would be better. Changed